parsl/tests/run.sh

```diff
@@ -0,0 +1,52 @@
+export PARSL_TESTING="true"
+
+set -e
+
+flake8 parsl/
+bash parsl/tests/lint-inits.sh
+
+# This uses all of the configurations and tests as the base from which to
+# run mypy checks - these are likely to capture most of the code used in
+# parsl
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/configs/
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/test*/
+MYPYPATH=$(pwd)/mypy-stubs mypy parsl/tests/sites/
+
+# do this before any testing, but not in-between tests
+rm -f .coverage
+
+pytest parsl -k "not cleannet" --config parsl/tests/configs/htex_local.py --cov=parsl --cov-append --cov-report= --random-order
+pytest parsl -k "not cleannet" --config parsl/tests/configs/local_threads.py --cov=parsl --cov-append --cov-report= --random-order
+
+# some of the site/ tests require more dependencies. These are installed here as needed,
+# so that the above tests happen with only the basic requirements installed.
+
+# workqueue
+./parsl/executors/workqueue/install-workqueue.sh
+export PATH=$PATH:/tmp/cctools/bin
+export PYTHONPATH=/tmp/cctools/lib/python3.5/site-packages
+
+# mpi
+bash parsl/executors/extreme_scale/install-mpi.sh $MPI
+if [[ "$MPI" == "mpich"   ]]; then mpichversion; fi
+if [[ "$MPI" == "openmpi" ]]; then ompi_info;    fi
+
+pip install .[extreme_scale,monitoring]
```

Comment:
if this `pip install` is happening in this file, then the original `pip install .` in the `install` section at line 24 should probably move into this file too.

Reply:
possibly if a user is running with `pip install -e` then having any kind of `pip install` in this script is going to be frustrating. see parameterisation comment for line 25?